plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageTestServer.java

```diff
@@ -57,44 +59,84 @@
     /** Request handlers for the requests made by the Google Cloud Storage client **/
     private final PathTrie<RequestHandler> handlers;
 
+    /** Server endpoint **/
+    private final String endpoint;
+
     /**
      * Creates a {@link GoogleCloudStorageTestServer} with the default endpoint
      */
     GoogleCloudStorageTestServer() {
-        this("https://www.googleapis.com", true);
+        this("https://www.googleapis.com");
     }
 
     /**
-     * Creates a {@link GoogleCloudStorageTestServer} with a custom endpoint,
-     * potentially prefixing the URL patterns to match with the endpoint name.
+     * Creates a {@link GoogleCloudStorageTestServer} with a custom endpoint
      */
-    GoogleCloudStorageTestServer(final String endpoint, final boolean prefixWithEndpoint) {
-        this.handlers = defaultHandlers(endpoint, prefixWithEndpoint, buckets);
+    GoogleCloudStorageTestServer(final String endpoint) {
```

Comment:
Long term (not in this PR) I think it would be good to move this into the fixture, and use a more pure unit testing mock for the service calls inside the unit tests.